lib/features/caching/caching_manager.dart

```diff
@@ -113,9 +113,12 @@ class CachingManager {
 
   Future<void> onUpgradeCache(int oldVersion, int newVersion) async {
     log('CachingManager::onUpgradeCache():oldVersion $oldVersion | newVersion: $newVersion');
-    await clearData();
-    if (oldVersion > 0 && oldVersion < newVersion && newVersion == 7) {
+    if (oldVersion > 0 &&
+        oldVersion < newVersion &&
+        (newVersion == 7 || newVersion == 10)) {
       await clearAll();
+    } else {
+      await clearData();
```

Comment:
we should refactor it:
- Create abstract class `MigrateSteps`
- Extend this class when we need, base on the condition. Fx: `MigrateV7`. `MigrateV10`.
- All the needed conditions, please move inside each implemented class.
Then we wil have a clear code here
```
Future<void> onUpgradeCache(int oldVersion, int newVersion) async {
    log('CachingManager::onUpgradeCache():oldVersion $oldVersion | newVersion: $newVersion');
    MigrateV7().execute();
    MigrateV10().execute();
    // ... 
}
```
WDYT? @dab246 
cc @nqhhdev @sherlockvn

Reply:
IMO, this separation is quite reasonable. But it should be named differently. Because we will have many types of databases later. So the version will be different.
- abstraction: `UpgradeDatabaseSteps`.
- Hivedatabase: `UpgradeHiveDatabaseStepsV10`